src/main/java/com/epam/internship/carrental/car/CarController.java

```diff
@@ -0,0 +1,294 @@
+package com.epam.internship.carrental.car;
+
+import com.epam.internship.carrental.car.enums.CarGearbox;
+import com.epam.internship.carrental.car.enums.CarType;
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+/**
+ * CarController providing a REST Api endpoint.
+ */
+@Api(tags="Cars")
+@Controller
+@RequestMapping(path = "/api/v1")
+public class CarController {
+
+    /**
+     * This field stores the instance of a CarService.
+     */
+    private final CarServiceImpl carService;
+
+    /**
+     * Autowired constructor for the class.
+     *
+     * @param carService service which provides access to the service layer.
+     */
+    @Autowired
+    public CarController(final CarServiceImpl carService) {
+        this.carService = carService;
+    }
+
+    /**
+     * Adds a new Car to the database, where the Car is defined in the request parameters.
+     * <pre>
+     * Method GET
+     * URL /api/v1/add
+     * </pre>
+     * Sample call: /api/v1/add?make=Dacia&model=1310&carType=Sedan&seats=5&fuelUsage=12.7&carGearbox=Manual
+     *
+     * @param make       maker of the car to be added.
+     * @param model      model of the car to be added.
+     * @param carType    carType of the car to be added.
+     * @param seats      number of seats in the car to be added.
+     * @param fuelUsage  fuel usage in liter/100km of the car to be added.
+     * @param carGearbox gearbox of the car to be added.
+     * @return ResponseEntity with Response Code 200 on success.
+     */
+    @GetMapping(path = "/add")
+    public @ResponseBody
```

Comment:
The style of the method signatures is somewhat personal preference unless it's not strictly defined, but I still propose that you change it here and everywhere in the code so that `public @ResponseBody ResponseEntity addNewCar` remains in the same line increasing the readability of the code.